numba/parfors/array_analysis.py

```diff
@@ -3148,7 +3185,19 @@ def _make_assert_equiv(self, scope, loc, equiv_set, _args, names=None):
             ir.Assign(value=value, target=var, loc=loc),
         ]
 
-    def _gen_shape_call(self, equiv_set, var, ndims, shape, post):
+    def _get_typ_part(self, typ, i):
+        if isinstance(typ, types.containers.UniTuple):
+            return typ.dtype
+        elif isinstance(typ, types.ArrayCompatible):
+            return types.intp
+        elif isinstance(typ, types.Number):
+            return typ
+        else:
+            if config.DEBUG_ARRAY_OPT >= 1:
+                print(1, "Unhandled case in _get_typ_part:", typ, i)
+            return types.intp
```

Comment:
This will end up in this case:
```python
from numba import njit
import numpy as np

@njit(parallel=True)
def foo(*args):
    x = np.random.randn(*args)
    return x

print(foo(np.int32(1), np.int64(2)))
```
it manifests as a lowering error:
```
TypeError: Can only insert i64 at [0] in [2 x i64]: got i32
```